src/DateTimePeriod.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+/*
+ * This file is part of Business.
+ *
+ * (c) Florian Voutzinos <florian@voutzinos.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Business;
+
+/**
+ * DateTime period implementation.
+ *
+ * @author Márk Sági-Kazár <mark.sagikazar@gmail.com>
+ */
+final class DateTimePeriod implements \IteratorAggregate
+{
+    private $datePeriod;
+
+    /**
+     * Creates a new DateTime period.
+     *
+     * @param \DateTime $startDate
+     * @param \DateTime $endDate
+     */
+    public function __construct(\DateTime $startDate, \DateTime $endDate)
+    {
+        $endDate = clone $endDate;
+        $endDate->modify('+1 day');
+
+        if ($startDate >= $endDate) {
+            throw new \InvalidArgumentException('Start date must be earlier than end date.');
```

Comment:
In `Business` object `LogicException` is thrown for the same case. Should I throw that one here as well?